pkg/gen/python-templates/kind.py.mustache

```diff
@@ -0,0 +1,44 @@
+import pulumi
+import pulumi.runtime
+
+class {{Kind}}(pulumi.CustomResource):
+    {{{Comment}}}
+    def __init__(self, __name__, __opts__=None{{#Properties}}, {{Name}}=None{{/Properties}}):
+        if not __name__:
+            raise TypeError('Missing resource name argument (for URN creation)')
+        if not isinstance(__name__, basestring):
+            raise TypeError('Expected resource name to be a string')
+        if __opts__ and not isinstance(__opts__, pulumi.ResourceOptions):
+            raise TypeError('Expected resource options to be a ResourceOptions instance')
+
+        __props__ = dict()
+
+        __props__['apiVersion'] = '{{RawAPIVersion}}'
+        self.apiVersion = '{{RawAPIVersion}}'
+
+        __props__['kind'] = '{{Kind}}'
+        self.kind = '{{Kind}}'
+        {{#RequiredProperties}}
+
+        if not {{Name}}:
+            raise TypeError('Missing required property {{Name}}')
+        elif not isinstance({{Name}}, {{PropType}}):
+            raise TypeError('Expected property aliases to be a {{PropType}}')
+        self.{{Name}} = {{Name}}
+        {{{Comment}}}
+        __props__['{{Name}}'] = {{Name}}
+        {{/RequiredProperties}}
+        {{#OptionalProperties}}
+
+        if {{Name}} and not isinstance({{Name}}, {{PropType}}):
+            raise TypeError('Expected property aliases to be a {{PropType}}')
+        self.{{Name}} = {{Name}}
```

Comment:
Same here, this is covered by the `__props__` right below

Reply:
or wait... maybe you do? I know that I'm working on populating properties the "node" way (a way that doesn't require this).

If you do have to set attributes on self, you should use self.__setattr__(name, {{Name}}) instead so we don't get syntax errors if any property names collide with keywords.

Reply:
I'm pretty much copying the AWS provider here. I actually have very little knowledge of what this _should_ look like.

Reply:
I would just copy TF providers. We can then improve both of them similarly.